specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2023-08-02-preview/JavaApps.json

```diff
@@ -0,0 +1,601 @@
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Start an App": {
+            "$ref": "./examples/v2/JavaApps/JavaApps_Start.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AppPlatform/javaApps/{appName}/stop": {
+      "post": {
+        "tags": [
+          "JavaApps"
+        ],
+        "summary": "Stop a JavaApp",
+        "operationId": "JavaApps_Stop",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/AppNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Stop operation is in progress.",
+            "headers": {
+              "Location": {
+                "type": "string"
+              }
+            }
+          },
+          "204": {
+            "description": "App stopped successfully."
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Stop an App": {
+            "$ref": "./examples/v2/JavaApps/JavaApps_Stop.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "JavaAppsCollection": {
+      "description": "JavaApps collection ARM resource.",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/JavaApp"
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "JavaApp": {
+      "description": "Azure Spring Apps JavaApp resource",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../app/resource-manager/Microsoft.App/preview/2023-04-01-preview/ContainerApps.json#/definitions/ContainerApp"
```

Comment:
I think it's not a good practice to reference definitions from another RP, it might introduce unexpected changes.

Reply:
Offline discussed with the author, this is not an issue on the ARM side.